### DISCUSSION TITLE
Some questions about lighthouse dns?

### DISCUSSION THREAD

**Discussion by zcz-1020** [submariner-io/lighthouse#1018] [Q&A] [answered]
Recently, I was looking at the lighthouse dns source-code and had some questions about the **getClusterIPForSvc** method of **record.go**。
```
func (lh *Lighthouse) getClusterIPForSvc(pReq *recordRequest) (*serviceimport.DNSRecord, bool) {
	localClusterID := lh.ClusterStatus.LocalClusterID()

	record, found, isLocal := lh.ServiceImports.GetIP(pReq.namespace, pReq.service, pReq.cluster, localClusterID, lh.ClusterStatus.IsConnected,
		lh.EndpointsStatus.IsHealthy)

	getLocal := isLocal || pReq.cluster != "" && pReq.cluster == localClusterID
	if found && getLocal {
		record, found = lh.LocalServices.GetIP(pReq.service, pReq.namespace)
	}

	return record, found
}
```
When both **found** and **getLocal** are true (that is, the serviceimport service on the local cluster is parsed) . Why do we still need to use **localServices.GetIP** to parse the IP address on the local cluster?

**Comment by sridhargaddam** [comment, marked as answer]
When an exported service is present even on the local cluster, Lighthouse will give local service precedence over the remote service.

**Reply by sridhargaddam** [reply]
You can refer to this PR for more details - https://github.com/submariner-io/lighthouse/issues/355

**Reply by zcz-1020** [reply]
Thank you. I see. I didn't think about globalnet is enabled before.
